File: infra/src/main/java/com/movie/infra/common/response/ApiResponse.java
package com.movie.infra.common.response;

import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class ApiResponse<T> {
    private final int code;
    private final String message;
    private final T data;

    private ApiResponse(int code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(HttpStatus.OK.value(), "Success", data);
    }

    public static <T> ApiResponse<T> error(HttpStatus status, String message) {
        return new ApiResponse<>(status.value(), message, null);
    }

    public static <T> ApiResponse<T> error(int code, String message) {
        return new ApiResponse<>(code, message, null);
    }
} 

File: module-common/src/main/kotlin/com/example/redis/response/ApiResponse.kt
package com.example.redis.response

import org.springframework.http.HttpStatus
import java.time.LocalDateTime

class ApiResponse<T>(
    status: HttpStatus,
    message: String,
    data: T? = null,
    errors: MutableList<ErrorResponse> = mutableListOf(),
    timestamp: LocalDateTime = LocalDateTime.now(),
) {
}